semeio/workflows/localisation/local_script_lib.py

```diff
@@ -0,0 +1,482 @@
+                data_size = gen_data_config.get_initial_size()
+                if data_size > 0:
+                    activate_gen_param(
+                        model_param_group, node_name, param_list, data_size
+                    )
+                else:
+                    debug_print(
+                        f"Parameter {node_name} has data size: {data_size} "
+                        f"in {ministep_name}",
+                        LogLevel.LEVEL3,
+                    )
+            elif impl_type == ErtImplType.SURFACE:
+                if corr_spec.surface_scale is not None:
+                    surface_file = corr_spec.surface_scale.surface_file
+                    debug_print(
+                        f"Get surface size from: {surface_file}",
+                        LogLevel.LEVEL3,
+                    )
+                    surface = Surface(surface_file)
+                    data_size = surface.getNX() * surface.getNY()
+                    activate_and_scale_correlation(
+                        surface,
+                        corr_spec.surface_scale.method,
+                        model_param_group.row_scaling(node_name),
+                        corr_spec.ref_point,
+                        data_size,
+                        corr_spec.surface_scale.main_range,
+                        corr_spec.surface_scale.perp_range,
+                        corr_spec.surface_scale.azimuth,
+                    )
+                else:
+                    debug_print(
+                        f"No correlation scaling for node {node_name} "
+                        f"in {ministep_name}",
+                        LogLevel.LEVEL3,
+                    )
+
+        # Setup observation group
+        for obs_name in obs_list:
+            debug_print(f"Add obs node: {obs_name}", LogLevel.LEVEL2)
+            obs_group.addNode(obs_name)
+
+        # Setup ministep
+        debug_print(
+            f"Attach {param_group_name} to ministep {ministep_name}", LogLevel.LEVEL1
+        )
+        ministep.attachDataset(model_param_group)
+
+        debug_print(
+            f"Attach {obs_group_name} to ministep {ministep_name}", LogLevel.LEVEL1
+        )
+        ministep.attachObsset(obs_group)
+
+        debug_print(f"Add {ministep_name} to update step\n", LogLevel.LEVEL1)
+        ert_local_config.getUpdatestep().attachMinistep(ministep)
+
+
+def clear_correlations(ert):
+    local_config = ert.getLocalConfig()
+    local_config.clear()
+
+
+def check_if_ref_point_in_grid(ref_point, grid):
+    try:
+        grid.find_cell_xy(ref_point[0], ref_point[1], 0)
+    except ValueError as err:
+        raise ValueError(
+            f"Reference point {ref_point} corresponds to undefined grid cell "
+            f"or is outside the area defined by the grid {grid.get_name()}\n"
+            "Check specification of reference point."
+        ) from err
+
+
+# This is an example callable which decays as a gaussian away from a position
```

Comment:
Is this comment correct? If it is intended it should probably be in the doc string instead?

Reply:
I removed it. Belongs more to docstring, yes.